maxfw/core/utils.py

```diff
@@ -15,21 +15,22 @@ def inner(*args, **kwargs):
             # run the function
             return func(*args, **kwargs)
         except ValueError as ve:
-            if 'specific_message' in str(ve):
-                raise NotImplementedError
+            if 'channel' in str(ve):
+                abort(400, "Invalid input, please ensure the input has right number of channels.")
+            elif 'mode' in str(ve):
+                abort(400, "Invalid input, please ensure the right mode is used with the given channels.")
             else:
-                raise NotImplementedError
+                abort(400, str(ve))
```

Comment:
Same issue here: avoid exposing exception information to the client

Reply:
So should I just return no message/dummy message along with the exception? 
the other option is to go through each of the 28 exceptions and write wrapper exceptions (more user friendly?) but that is extremely redundant.

Reply:
I think you can send a more generic message, and log the exception internally.